lib/install/management/appliance.go

```diff
@@ -118,8 +118,61 @@ func (d *Dispatcher) isContainerVM(vm *vm.VirtualMachine) (bool, error) {
 
 func (d *Dispatcher) checkExistence(conf *config.VirtualContainerHostConfigSpec, settings *data.InstallerData) error {
 	defer trace.End(trace.Begin(""))
-
 	var err error
+
+	// We should check in the folder first then in the resource pool. If a folder collision occurs we want to fail as soon as possible.
+	folderRef, err := VchFolder(d.op, d.session, conf)
+	if err != nil {
+		// log it and continue, we would expect the folder to not exist in a fresh install
+		d.op.Debugf("VCH folder did not exist during existence check, received error: %s", err)
+	}
+
+	if folderRef != nil {
+		d.op.Debugf("Found existing VCH Folder(%s), inspecting contents.", folderRef.InventoryPath)
+		// check for existence of a vch or vm with the vch name
+		children, err := folderRef.Children(d.op)
+		if err != nil {
+			// log the error and fail, we cannot confirm at this point if a vch exists.
+			d.op.Debugf("Could not collect the children of the VCH Folder: %s", err)
+			return err
+		}
+
+		d.op.Debugf("Found Children of target VCH Folder: %s", children)
+		for _, child := range children {
+			// fetch the object from vsphere
+			obj, err := d.session.Finder.ObjectReference(d.op, child.Reference())
+			if err != nil {
+				// what to do... skip for now
```

Comment:
@cgtexmex I did have some questions here, and the comment right below this one as to the procedure. I think we should actually fail since we want to fail early rather than attempt to start making things only to find the vm did exist.

Reply:
I don't think this is needed -- if there's already an object with this name the `CreateFolder` call will fail -- there will be no additional create calls attempted and nothing would be orphaned.  If a CI test is failing because of that error message then the test needs updated.

Reply:
I see, yeah all of that is pretty much overly defensive. Thanks! It has been removed and a simpler error message is now returned.